silverback/cluster/types.py

```diff
@@ -0,0 +1,281 @@
+import enum
+import math
+import uuid
+from datetime import datetime
+from typing import Annotated
+
+from pydantic import BaseModel, Field, computed_field, field_validator
+
+
+class WorkspaceInfo(BaseModel):
+    id: uuid.UUID
+    owner_id: uuid.UUID
+    name: str
+    slug: str
+
+
+class ClusterConfiguration(BaseModel):
+    """Configuration of the cluster (represented as 16 byte value)"""
+
+    # NOTE: This configuration must be encode-able to a uint64 value for db storage
+    #       and on-chain processing through ApePay
+
+    # NOTE: All defaults should be the minimal end of the scale,
+    #       so that `__or__` works right
+
+    # Version byte (Byte 0)
+    # NOTE: Just in-case we change this after release
+    version: int = 1
+
+    # Bot Worker Configuration (Bytes 1-2)
+    cpu: Annotated[int, Field(ge=0, le=16)] = 0  # 0.25 vCPU
+    """Allocated vCPUs per bot: 0.25 vCPU (0) to 16 vCPU (6)"""
```

Comment:
question: I'm confused by this. 6 steps from 0.25 to 16?

Reply:
I got this from AWS docs

Reply:
Got what from AWS docs? My question is about your encoding.

Reply:
The encoding is internal, when it's displayed it shows up as `x.xx vCPU` to be human-readable

Reply:
I was just asking a question.

Reply:
The encoding is non-linear, yes

Reply:
ah, I see what you were saying. fixed the typo

Reply:
the value is  `f"{0.25 ** cpu} vCPU"`